cohere/generation.py

```diff
@@ -1,33 +1,70 @@
-from cohere.response import CohereObject
-from typing import List
+from concurrent.futures import Future
+from typing import Any, Dict, Iterator, List, NamedTuple, Optional, Union
 
+from cohere.response import AsyncAttribute, CohereObject
 
-class TokenLikelihood(CohereObject):
-    def __init__(self, token: str, likelihood: float) -> None:
-        self.token = token
-        self.likelihood = likelihood
+TokenLikelihood = NamedTuple("TokenLikelihood", [("token", str), ("likelihood", float)])
 
 
-class Generation(CohereObject):
-    def __init__(self,
-                 text: str,
-                 likelihood: float,
-                 token_likelihoods: List[TokenLikelihood]) -> None:
+class Generation(CohereObject, str):
+
+    def __new__(cls, text: str, *_, **__):
+        return str.__new__(cls, text)
+
+    def __init__(self, text: str, likelihood: float, token_likelihoods: List[TokenLikelihood]) -> None:
         self.text = text
         self.likelihood = likelihood
         self.token_likelihoods = token_likelihoods
 
+    def __str__(self) -> str:
+        return str(self.text)
+
+    def __len__(self) -> int:
+        return len(self.text)
+
+    def __getitem__(self, key):
+        return self.text[key]
+
 
 class Generations(CohereObject):
+
     def __init__(self,
-                 generations: List[Generation],
-                 return_likelihoods: str) -> None:
-        self.generations = generations
+                 return_likelihoods: str,
+                 response: Optional[Dict[str, Any]] = None,
+                 *,
+                 _future: Optional[Future] = None) -> None:
+        self.generations: Union[AsyncAttribute, List[Generation]] = None
         self.return_likelihoods = return_likelihoods
-        self.iterator = iter(generations)
+        if _future is not None:
+            self._init_from_future(_future)
+        else:
+            assert response is not None
+            self.generations = self._generations(response)
+
+    def _init_from_future(self, future: Future):
+        self.generations = AsyncAttribute(future, self._generations)
+
+    def _generations(self, response: Dict[str, Any]) -> List[Generation]:
+        generations: List[Generation] = []
+        for gen in response['generations']:
+            likelihood = None
+            token_likelihoods = None
+            if self.return_likelihoods == 'GENERATION' or self.return_likelihoods == 'ALL':
```

Comment:
is this more pythonic? 
```suggestion
            if if self.return_likelihoods in ['GENERATION', 'ALL']:
```